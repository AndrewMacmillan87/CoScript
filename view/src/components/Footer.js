import React, { Fragment } from "react";
import { Link, withRouter } from "react-router-dom";

// Redux
import { connect } from "react-redux";
import { getMostRecentProject } from "../redux/actions/documents";

import PropTypes from "prop-types";

import "./footer.css";

export const Footer = ({ user, loading, getMostRecentProject, ...props }) => {
    const authLinks = (
        <div className='footer-links'>
            <Link to='/'>Home</Link>
            <Link to='/profile'>Profile</Link>
            <Link
                to='#'
                onClick={async () => {
                    await getMostRecentProject();
                    const { projectData } = props;
                    projectData
                        ? props.history.push(`/editor${projectData.url}`)
                        : props.history.push("/editor");
                }}>
                Workspace
            </Link>
            <Link to='/projects'>Active Projects</Link>
            <Link to='/publications'>Publications</Link>
        </div>
    );

    const guestLinks = (
        <div className='footer-links'>
            <Link to='/'>Home</Link>
            <Link to='/projects'>Active Projects</Link>
            <Link to='/publications'>Publications</Link>
        </div>
    );

    const isLoggedIn = user ? user.isLoggedIn : false;

    return (
        <footer className='footer'>
            <h2 className='footer-h2'>CoScript</h2>
            <div className='footer-container'>
                <div className='links-container socials'>
                    <p>Socials</p>
                    <div className='footer-links'>
                        <a href='https://github.com/AndrewMacmillan87/coscript'>
                            GitHub
                        </a>
                        <a href='/'>Twitter</a>
                        <a href='/'>Facebook</a>
                    </div>
                </div>
                <div className='links-container site-map'>
                    <p>Site Map</p>

                    {!loading && (
                        <Fragment>
                            {isLoggedIn ? authLinks : guestLinks}
                        </Fragment>
                    )}
                </div>
                <div className='links-container legals'>
                    <p>Legal</p>
                    <div className='footer-links'>
                        <a href='/terms'>Terms and Conditions</a>
                        <a href='/copyright'>Copyright Info</a>
                        <a href='/agreement'>Writers Agreements</a>
                        <a href='/'>Accreditation</a>
                    </div>
                </div>
            </div>
        </footer>
    );
};

Footer.propTypes = {
    user: PropTypes.object,
    loading: PropTypes.bool,
    getMostRecentProject: PropTypes.func.isRequired,
};

const mapProps = (state) => ({
    user: state.auth.user,
    loading: state.auth.loading,
    projectData: state.documents.projectData,
});

export default connect(mapProps, { getMostRecentProject })(withRouter(Footer));
