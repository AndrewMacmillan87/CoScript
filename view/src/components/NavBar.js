import React, { Fragment } from "react";
import { Link, withRouter } from "react-router-dom";
import Button from "./buttons/Button";

import { getUserMessages } from "../redux/auxiliary/user-messages";

// Redux
import { connect } from "react-redux";
import { getMostRecentProject } from "../redux/actions/documents";

import PropTypes from "prop-types";

import "./navbar.css";

export class NavBar extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            visible: false,
            unread: false,
        };
        this.handleClick = this.handleClick.bind(this);
    }

    /**
     * Checks to see if the user has any unread messages
     * in their personal profile message inbox and sets
     * the 'unread' state property to true if they have.
     * This displays a small icon to the user to indicate
     * they have unread messages
     */
    async componentDidMount() {
        const userMessages = await getUserMessages();
        for (let message of userMessages) {
            if (!message.read) {
                this.setState({
                    unread: true,
                });
            }
        }
    }

    /**
     * Sets the 'visible' state property so that the
     * dropdown menu can be revealed and hidden by
     * clicking the link
     */
    handleClick() {
        this.setState((state) => ({
            visible: !state.visible,
        }));
    }

    render() {
        const { unread } = this.state;
        const { user, loading, history } = this.props;

        const menuIcon = require("../img/menu.svg");
        const spinner = require("../img/loading.svg");
        const isLoggedIn = user && user.isLoggedIn | false;

        const loggedIn = (
            <Fragment>
                <div className='header'>
                    <div className='logo-container'>
                        <Link to='/'>
                            <p className='logo'>CoScript</p>
                        </Link>
                    </div>
                    <div className='nav-links'>
                        <Link to='/projects'>Active Projects</Link>
                        <Link to='/publications'>Publications</Link>
                        <Link
                            to='#'
                            onClick={async () => {
                                await this.props.getMostRecentProject();
                                const { projectData } = this.props;
                                projectData
                                    ? history.push(`/editor${projectData.url}`)
                                    : history.push("/editor");
                            }}>
                            Workspace
                        </Link>
                    </div>
                </div>
                <div className='nav-btns'>
                    {user && !loading ? (
                        <div className='logged-in-user-container'>
                            <p className='user-username'>
                                {user.profile.displayName}
                            </p>
                            <button
                                className='desktop-icon-link'
                                onClick={this.handleClick}>
                                {unread ? (
                                    <div className='avatar-unread-msg-container'>
                                        <div className='avatar-unread-msg-icon'></div>
                                        <img
                                            className='user-gravatar'
                                            src={user.avatar}
                                            alt='user gravatar'
                                        />
                                    </div>
                                ) : (
                                    <img
                                        className='user-gravatar'
                                        src={user.avatar}
                                        alt='user gravatar'
                                    />
                                )}
                            </button>
                        </div>
                    ) : (
                        <img
                            className='user-spinner'
                            src={spinner}
                            alt='loading spinner icon'
                        />
                    )}
                </div>
                <button
                    className='dropdown-icon-link'
                    onClick={this.handleClick}>
                    <img
                        className='dropdown-icon'
                        src={menuIcon}
                        alt='Burger menu icon'
                    />
                </button>
                <div
                    id='dropdown-nav'
                    className={
                        this.state.visible
                            ? "dropdown-menu dropdown-content display"
                            : "dropdown-menu dropdown-content"
                    }>
                    <Link to='/projects'>
                        <p className='dropdown-link'>Active Projects</p>
                    </Link>
                    <Link to='/publications'>
                        <p className='dropdown-link'>Publications</p>
                    </Link>
                    <Link
                        to='#'
                        onClick={async () => {
                            await this.props.getMostRecentProject();
                            const { projectData } = this.props;
                            projectData
                                ? history.push(`/editor${projectData.url}`)
                                : history.push("/editor");
                        }}>
                        <p className='dropdown-link'>Workspace</p>
                    </Link>
                    <Link to='/profile'>
                        <p className='dropdown-link'>My Profile</p>
                    </Link>
                    <a
                        onClick={() => {
                            history["from"] = "logout";
                            history.push("/redirect");
                        }}
                        href=' '>
                        <p className='dropdown-link'>Logout</p>
                    </a>
                </div>
            </Fragment>
        );

        const visiting = (
            <Fragment>
                <div className='header'>
                    <div className='logo-container'>
                        <Link to='/'>
                            <p className='logo'>CoScript</p>
                        </Link>
                    </div>
                    <div className='nav-links'>
                        <Link to='/projects'>Active Projects</Link>
                        <Link to='/publications'>Publications</Link>
                    </div>
                </div>
                <div className='nav-btns'>
                    <div>
                        <Link to='/register'>
                            <Button classes='btn' value='Register' />
                        </Link>
                    </div>
                    <div>
                        <Link to='/login'>
                            <Button classes='grn-btn' value='Login' />
                        </Link>
                    </div>
                </div>
                <button
                    className='dropdown-icon-link'
                    onClick={this.handleClick}>
                    <img
                        className='dropdown-icon'
                        src={menuIcon}
                        alt='Burger menu icon'
                    />
                </button>
                <div
                    id='dropdown-nav'
                    className={
                        this.state.visible
                            ? "dropdown-menu dropdown-content display"
                            : "dropdown-menu dropdown-content"
                    }>
                    <Link to='/register'>
                        <p className='dropdown-link'>Register</p>
                    </Link>
                    <Link to='/login'>
                        <p className='dropdown-link'>Login</p>
                    </Link>
                    <Link to='/projects'>
                        <p className='dropdown-link'>Active Projects</p>
                    </Link>
                    <Link to='/publications'>
                        <p className='dropdown-link'>Publications</p>
                    </Link>
                </div>
            </Fragment>
        );

        return (
            <nav className='navbar'>
                {!loading && (
                    <Fragment>{!isLoggedIn ? visiting : loggedIn}</Fragment>
                )}
            </nav>
        );
    }
}

NavBar.propTypes = {
    user: PropTypes.object,
    loading: PropTypes.bool,
    getMostRecentProject: PropTypes.func.isRequired,
};

const setProps = (state) => ({
    user: state.auth.user,
    loading: state.auth.loading,
    projectData: state.documents.projectData,
});

export default connect(setProps, {
    getMostRecentProject,
})(withRouter(NavBar));
