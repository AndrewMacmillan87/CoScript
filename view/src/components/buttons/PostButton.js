import React from "react";
import PropTypes from "prop-types";

import "./buttons.css";

const PostButton = ({ onClick, classes, value }) => {
    return (
        <input
            onClick={onClick}
            className={classes}
            type='submit'
            value={value}
        />
    );
};

PostButton.defaultProps = {
    value: "Verify"
};

PostButton.propTypes = {
    onClick: PropTypes.func,
    classes: PropTypes.string,
    value: PropTypes.string
};

export default PostButton;
