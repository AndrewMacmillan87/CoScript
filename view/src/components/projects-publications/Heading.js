import React, { useRef } from "react";
import { withRouter } from "react-router-dom";

import PostButton from "../buttons/PostButton";

import PropTypes from "prop-types";

import { filterByCriteria } from "../../redux/auxiliary/projects";

// Redux
import { connect } from "react-redux";
import { loadAllProjects } from "../../redux/actions/documents";

export const Heading = ({
    banner,
    bannerHeading,
    searchPlaceholder,
    searchCriteria,
    onChange,
    setProjects,
    ...props
}) => {
    const searchBar = useRef(null);

    /**
     * Calls the 'setProjects' function with the appropriate
     * parameters based on the search filter selected by the
     * user. Only projects with any matches to the search
     * criteria are displayed after the search results come
     * back
     *
     * @param {Object} e - event object
     */
    const onSubmit = async (e) => {
        e.preventDefault();
        await props.loadAllProjects();
        const status =
            props.location.pathname === "/projects" ? "active" : "published";
        switch (searchBar.current.value) {
            case "author":
                setProjects(
                    filterByCriteria(
                        props.projectArray,
                        status,
                        searchCriteria,
                        "author"
                    )
                );
                break;
            case "title":
                setProjects(
                    filterByCriteria(
                        props.projectArray,
                        status,
                        searchCriteria,
                        "title"
                    )
                );
                break;
            case "genre":
                setProjects(
                    filterByCriteria(
                        props.projectArray,
                        status,
                        searchCriteria,
                        "genres"
                    )
                );
                break;
            default:
                return;
        }
    };

    return (
        <header className='banner-container'>
            <div id={banner} className='banner-heading-container'>
                <h1 className='banner-heading'>{bannerHeading}</h1>
            </div>
            <div className='search-bar-container'>
                <div className='search-bar'>
                    <form className='search-form' onSubmit={(e) => onSubmit(e)}>
                        <input
                            className='search-input'
                            type='text'
                            name='search-input'
                            onChange={(e) => onChange(e)}
                            placeholder={searchPlaceholder}
                            onFocus={(e) => (e.target.placeholder = "")}
                            onBlur={(e) =>
                                (e.target.placeholder = searchPlaceholder)
                            }
                        />
                        <PostButton classes='square-btn' value='Search' />
                        <select className='sort-by-selector' ref={searchBar}>
                            <option name='author' value='author'>
                                Author
                            </option>
                            <option name='title' value='title'>
                                Title
                            </option>
                            <option name='genre' value='genre'>
                                Genre
                            </option>
                        </select>
                    </form>
                </div>
            </div>
        </header>
    );
};

Heading.propTypes = {
    banner: PropTypes.string.isRequired,
    bannerHeading: PropTypes.string.isRequired,
    searchPlaceholder: PropTypes.string.isRequired,
    searchCriteria: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    setProjects: PropTypes.func.isRequired,
    loadAllProjects: PropTypes.func.isRequired,
    projectArray: PropTypes.array,
};

const mapProps = (state) => ({
    projectArray: state.documents.projectArray,
});

export default connect(mapProps, { loadAllProjects })(withRouter(Heading));
