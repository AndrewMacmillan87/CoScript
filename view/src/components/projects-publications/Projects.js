import React, { Fragment, useState, useEffect } from "react";
import { withRouter } from "react-router-dom";

import NavBar from "../NavBar";
import Heading from "./Heading";
import Footer from "../Footer";
import ProjectCard from "../cards/ProjectCard";

// Redux
import { connect } from "react-redux";
import { loadProject, loadAllProjects } from "../../redux/actions/documents";
import { clearFileData } from "../../redux/actions/files";
import { filterActive } from "../../redux/auxiliary/projects";
import { stopSoundEffect } from "../../redux/auxiliary/typing-sound-effect";

import PropTypes from "prop-types";

import "./pro-pub.css";

export const Projects = ({ ...props }) => {
    /**
     * Set state for this component and its children
     */
    const [projects, setProjects] = useState(null);
    const [loading, setLoading] = useState(true);
    const [searchCriteria, setSearchCriteria] = useState("");
    const [searching, setSearching] = useState(false);

    /**
     * Destructure this components props
     */
    const {
        loadProject,
        loadAllProjects,
        clearFileData,
        projectArray,
        history,
    } = props;

    /**
     * Scroll to the top of the
     * page on mount. Also load all project data into
     * application level state inside an async IIFE function
     * to avoid memory leaks if the component unmounts before
     * the API call can complete
     */
    useEffect(() => {
        window.scrollTo(0, 0);
        (async function load() {
            await loadAllProjects();
        })();
    }, [loadAllProjects]);

    /**
     * Sets the browser tab title and scrolls to the top
     * of the page when the component mounts. Also stops
     * playing te type writer sound effect if the user
     * navigated to this page from the introductory
     * workspace.
     *
     * A check is also performed on whether or not there
     * is any file data being persisted in Redux, if there
     * is, it is cleared so that the user can be shown the
     * introductory workspace if they have no personal
     * projects, or a recent personal project can be loaded
     * when they go back to the workspace
     *
     * All projects are then loaded into the frontend and
     * all active projects are filtered out and displayed
     */
    useEffect(() => {
        document.title = "CoScript - Active Projects";
        stopSoundEffect();
        clearFileData();
        const projects = projectArray ? projectArray : [];
        if (!searching) {
            setProjects(filterActive(projects));
        }
        setLoading(false);
    }, [clearFileData, projectArray, searching]);

    /**
     * Sets the 'searchCriteria' state property to the
     * value of the event target
     *
     * @param {Object} e - event object
     */
    const onChange = (e) => {
        setSearchCriteria(e.target.value);
    };

    /**
     * Sets the 'projects' state property to the
     * <projectArray> and also sets the 'searching'
     * state property to true
     *
     * @param {Array} searchResults - an array of projects
     */
    const updateProjects = (searchResults) => {
        setSearching(true);
        setProjects(searchResults);
    };

    const sky = require("../../img/sky-example.jpg");
    const spinner = require("../../img/loading-black.svg");

    return (
        <main>
            <NavBar />
            <Heading
                banner='projects-banner'
                bannerHeading='Active Projects'
                searchPlaceholder='Search for projects...'
                searchCriteria={searchCriteria}
                onChange={onChange}
                setProjects={updateProjects}
            />
            {loading ? (
                <div className='project-publication-spinner'>
                    <img
                        className='project-publication-spinner'
                        src={spinner}
                        alt='loading'
                    />
                </div>
            ) : (
                <Fragment>
                    {projects && projects.length ? (
                        <div className='project-cards-container'>
                            {projects.map((project) => {
                                return (
                                    <ProjectCard
                                        key={project.projectId}
                                        src={sky}
                                        img_alt='Project cover image'
                                        title={project.title}
                                        project_admin={project.author}
                                        genres={project.genres}
                                        onClick={async () => {
                                            await loadProject(project);
                                            history.push(
                                                `/landing${project.url}`
                                            );
                                        }}
                                    />
                                );
                            })}
                        </div>
                    ) : (
                        <Fragment>
                            {searching ? (
                                <h2 className='no-projects-msg'>
                                    No matches found
                                </h2>
                            ) : (
                                <h2 className='no-projects-msg'>
                                    There are currently no active projects
                                </h2>
                            )}
                        </Fragment>
                    )}
                </Fragment>
            )}
            <Footer />
        </main>
    );
};

Projects.propTypes = {
    loadProject: PropTypes.func.isRequired,
    loadAllProjects: PropTypes.func.isRequired,
    clearFileData: PropTypes.func.isRequired,
    projectArray: PropTypes.array,
};

const mapProps = (state) => ({
    projectArray: state.documents.projectArray,
});

export default connect(mapProps, {
    loadProject,
    loadAllProjects,
    clearFileData,
})(withRouter(Projects));
