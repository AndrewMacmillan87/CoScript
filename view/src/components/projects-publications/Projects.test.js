import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { Provider } from "react-redux";
import { reduxStore } from "../../redux/store/reduxStore";

import { Projects } from "./Projects";
import { Heading } from "./Heading";
import NavBar from "../NavBar";
import PostButton from "../buttons/PostButton";
import Footer from "../Footer";

describe("Projects component", () => {
    let props, mockClearFileData, mockLoadAllProjects;
    beforeEach(() => {
        global.HTMLMediaElement.prototype.pause = jest.fn();
        global.scrollTo = jest.fn();
        mockClearFileData = jest.fn();
        mockLoadAllProjects = jest.fn();
        props = {
            loadProject: jest.fn(),
            loadAllProjects: mockLoadAllProjects,
            clearFileData: mockClearFileData,
            projectArray: [],
        };
    });

    test("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(
            <Provider store={reduxStore}>
                <Router>
                    <Projects {...props} />
                </Router>
            </Provider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(
                <Provider store={reduxStore}>
                    <Router>
                        <Projects {...props} />
                    </Router>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render a main component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Projects {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("main").length).toEqual(1);
    });

    it("should render the NavBar component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Projects {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(NavBar).length).toEqual(1);
    });

    it("should render the Heading component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Projects {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(Heading).length).toEqual(1);
    });

    it("should render the Footer component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Projects {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(Footer).length).toEqual(1);
    });

    it("should call `clearFileData()` on mount", () => {
        mount(
            <Provider store={reduxStore}>
                <Router>
                    <Projects {...props} />
                </Router>
            </Provider>
        );
        expect(mockClearFileData).toBeCalled();
    });

    it("should call `loadAllProjects()` on mount", async () => {
        mount(
            <Provider store={reduxStore}>
                <Router>
                    <Projects {...props} />
                </Router>
            </Provider>
        );
        await expect(mockLoadAllProjects).toBeCalled();
    });
});

describe("Heading component", () => {
    let props;
    beforeEach(() => {
        props = {
            banner: "",
            bannerHeading: "",
            searchPlaceholder: "",
            searchCriteria: "",
            onChange: jest.fn(),
            setProjects: jest.fn(),
            loadAllProjects: jest.fn(),
            projectArray: [],
        };
    });

    it("should render a header element with a className of banner-container", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Heading {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("header").length).toEqual(1);
        expect(wrapper.find({ className: "banner-container" }).length).toEqual(
            1
        );
    });

    it("should render a form element", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Heading {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("form").length).toEqual(1);
    });

    it("should render 2 input elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Heading {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("input").length).toEqual(2);
    });

    it("should render a select element", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Heading {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("select").length).toEqual(1);
    });

    it("should render 3 option elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Heading {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("option").length).toEqual(3);
    });

    it("should render the PostButton component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Heading {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(PostButton).length).toEqual(1);
    });
});
