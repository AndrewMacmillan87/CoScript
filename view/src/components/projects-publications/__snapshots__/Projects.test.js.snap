// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects component renders correctly 1`] = `
<main>
  <nav
    className="navbar"
  />
  <header
    className="banner-container"
  >
    <div
      className="banner-heading-container"
      id="projects-banner"
    >
      <h1
        className="banner-heading"
      >
        Active Projects
      </h1>
    </div>
    <div
      className="search-bar-container"
    >
      <div
        className="search-bar"
      >
        <form
          className="search-form"
          onSubmit={[Function]}
        >
          <input
            className="search-input"
            name="search-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Search for projects..."
            type="text"
          />
          <input
            className="square-btn"
            type="submit"
            value="Search"
          />
          <select
            className="sort-by-selector"
          >
            <option
              name="author"
              value="author"
            >
              Author
            </option>
            <option
              name="title"
              value="title"
            >
              Title
            </option>
            <option
              name="genre"
              value="genre"
            >
              Genre
            </option>
          </select>
        </form>
      </div>
    </div>
  </header>
  <div
    className="project-publication-spinner"
  >
    <img
      alt="loading"
      className="project-publication-spinner"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "loading-black.svg",
        }
      }
    />
  </div>
  <footer
    className="footer"
  >
    <h2
      className="footer-h2"
    >
      CoScript
    </h2>
    <div
      className="footer-container"
    >
      <div
        className="links-container socials"
      >
        <p>
          Socials
        </p>
        <div
          className="footer-links"
        >
          <a
            href="https://github.com/AndrewMacmillan87/coscript"
          >
            GitHub
          </a>
          <a
            href="/"
          >
            Twitter
          </a>
          <a
            href="/"
          >
            Facebook
          </a>
        </div>
      </div>
      <div
        className="links-container site-map"
      >
        <p>
          Site Map
        </p>
      </div>
      <div
        className="links-container legals"
      >
        <p>
          Legal
        </p>
        <div
          className="footer-links"
        >
          <a
            href="/terms"
          >
            Terms and Conditions
          </a>
          <a
            href="/copyright"
          >
            Copyright Info
          </a>
          <a
            href="/agreement"
          >
            Writers Agreements
          </a>
          <a
            href="/"
          >
            Accreditation
          </a>
        </div>
      </div>
    </div>
  </footer>
</main>
`;
